gitRunner.go

```diff
@@ -154,13 +154,18 @@ func gitClone(ctx context.Context, gitName, gitURL, gitBranch string, shallowClo
 }
 
 func gitCheckout(ctx context.Context, gitRevision string) (err error) {
+	args := []string{"fetch", "origin", "--depth", "1", gitRevision}
```

Comment:
Do we just want to fetch the last commit id? May be its also good to specify a range of commits

```
args := []string{"fetch", "origin", "--depth", "1", gitRevision + "^..HEAD"}
```

Reply:
We just need to checkout to a revision. So we may not need to fetch too many commits.